tuxemon/core/states/world/worldstate.py

```diff
@@ -634,7 +666,8 @@ def get_exits(self, position, collision_map=None, skip_nodes=None):
                 ("left", (position[0] - 1, position[1])),
         ):
             # if exits are defined make sure the neighbor is present there
-            if exits and not neighbor in exits:
+            self_blocks = self.tile_blocks(entity, collision_map.get(position))
```

Comment:
The collision checking code must be very fast because it is used for pathfinding and player movement.  The reason why the collision map exists is so that complex collision logic is handled only once when the map is loaded.  As I understand, you are creating tiles which conditionally block depending on the string value assigned to them.  Since that is the case, you should move the tile text parsing to the code which builds the collision map.  Its not a good practice to use things like `text.split` in a tight loop.  Instead, you should break down the values and store in a list, which can be more quickly compared against.  I understand that this is a small change overall, but i want to make sure you understand that we cannot keep adding inefficient code into the collision detection code.  Its already a bottleneck for pathfinding, and I don't want to watch it get worse.

Reply:
That's very good to know, didn't think about it. I'll see how I can do that so the splitting is done only once. Will need to store the result in the collision system but I'm sure I'll find a way.

Reply:
The solution was very easy, will be fixed in the next commit.